build-support/bin/packages.py

```diff
@@ -60,25 +44,22 @@ def __str__(self):
     def __repr__(self):
         return f"Package<name={self.name}>"
 
-    def exists(self) -> bool:
-        req = Request(f"https://pypi.org/pypi/{self.name}")
-        req.get_method = lambda: "HEAD"  # type: ignore[assignment]
-        try:
-            urlopen(req)
+    def exists(self) -> bool:  # type: ignore[return]
```

Comment:
MyPy doesn't understand that `response.raise_for_status()` will raise an exception, so it complains that this method sometimes doesn't always return `bool`.